src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1625,66 +1647,105 @@ public void UpdateWithExtraWorkspaceViewInfo(ExtraWorkspaceViewInfo workspaceVie
                     nodeModel.UpdateValue(new UpdateValueParams("IsUpstreamVisible", nodeViewInfo.IsUpstreamVisible.ToString()));
                 }
             }
+        }
 
-            foreach (ExtraNoteViewInfo noteViewInfo in workspaceViewInfo.Notes)
+        private void LoadLegacyNotes(IEnumerable<ExtraNoteViewInfo> noteViews)
+        {
+            if (noteViews == null)
+              return;
+
+            foreach (ExtraNoteViewInfo noteViewInfo in noteViews)
             {
                 var guidValue = IdToGuidConverter(noteViewInfo.Id);
 
                 // TODO, QNTM-1099: Figure out if ZIndex needs to be set here as well
                 var noteModel = new NoteModel(noteViewInfo.X, noteViewInfo.Y, noteViewInfo.Text, guidValue);
                 this.AddNote(noteModel);
             }
+        }
+
+        private void LoadNotesFromAnnotations(IEnumerable<ExtraAnnotationViewInfo> annotationViews)
+        {
+            if (annotationViews == null)
+              return;
+
+            foreach (ExtraAnnotationViewInfo annotationViewInfo in annotationViews)
+            {
+                if (annotationViewInfo.Nodes == null)
```

Comment:
in what case nodes will be null?

Reply:
Is this for if user manually modified the file?

Reply:
I am not sure in what case Nodes would be null, it is just a habit of mine to check for null before using an object. It is also a habit of mine to split multiple if checks into multiple tests to ease debugging. I can remove the null check or combine the two if that is what the team wants, but I would recommend to not do this.

Reply:
well, nodes is an empty array and I don't think nodes will be null. But I agree to have some extra checks.